commit 100cc1d828547ec9019214d20e2d72de1ea2f6d1
Author: Trent Mick <trentm@gmail.com>
Date:   2019-05-10T12:58:27-07:00 (5 months ago)
    
    TRITON-1662 setup-prometheus.sh's loose package selection broken by packages with brand=bhyve
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/setup-grafana.sh b/setup-grafana.sh
index f09ede0..477c513 100755
--- a/setup-grafana.sh
+++ b/setup-grafana.sh
@@ -4,7 +4,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
-# Copyright (c) 2018 Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 #
@@ -69,7 +69,14 @@ headnode_uuid=\$(sysinfo | json UUID)
 admin_uuid=\$(sdc-useradm get admin | json uuid)
 admin_network_uuid=\$(sdc-napi /networks?name=admin | json -H 0.uuid)
 external_network_uuid=\$(sdc-napi /networks?name=external | json -H 0.uuid)
-package=\$(sdc-papi /packages | json -Ha uuid max_physical_memory | sort -n -k 2 \
+
+# Find package
+# - Exclude packages with "brand" set to limit to a provision of a particular
+#   brand (e.g. brand=bhyve ones that have shown up recently for flexible
+#   disk support).
+package=\$(sdc-papi /packages \
+    | json -c '!this.brand' -Ha uuid max_physical_memory \
+    | sort -n -k 2 \
     | while read uuid mem; do
 
     # Find the first one with at least ${MIN_MEMORY}
diff --git a/setup-prometheus.sh b/setup-prometheus.sh
index a9f1ad8..4351576 100755
--- a/setup-prometheus.sh
+++ b/setup-prometheus.sh
@@ -4,7 +4,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
-# Copyright (c) 2018 Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 #
@@ -83,7 +83,12 @@ network_uuid=\$(vmadm get \$(vmadm lookup alias=~^cmon | head -1) | json nics.1.
 admin_network_uuid=\$(sdc-napi /networks?name=admin | json -H 0.uuid)
 
 # Find package
-package=\$(sdc-papi /packages | json -Ha uuid max_physical_memory | sort -n -k 2 \
+# - Exclude packages with "brand" set to limit to a provision of a particular
+#   brand (e.g. brand=bhyve ones that have shown up recently for flexible
+#   disk support).
+package=\$(sdc-papi /packages \
+    | json -c '!this.brand' -Ha uuid max_physical_memory \
+    | sort -n -k 2 \
     | while read uuid mem; do
 
     # Find the first one with at least ${MIN_MEMORY}
